src/data/W/cardinal.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2021 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import set_theory.cardinal_ordinal
+import data.W.basic
+/-!
+# Cardinality of W-types
+
+This file proves some theorems about the cardinality of W-types. The main result is
+`cardinal_mk_le_max_omega_of_fintype` which says that if for any `a : α`,
+`β a` is finite, then the cardinality of `W_type β` is at most the maximum of the
+cardinality of `α` or `cardinal.omega`.
+This has applications in first order logic, and can be used to prove that in the
+cardinality of the set of terms in a language with is bound by the cardinality
+of the set of function and constant symbols if that set is infinite, and the set of terms is
+countable if there are finitely many constant and function symbols.
```

Comment:
Please break this sentence into two or three pieces. Also, `used to prove that in the` and `language with is bound by` seem to be unnatural/contain typos.

Reply:
I've completely changed this part.